zebrad/src/components/inbound/tests.rs

```diff
@@ -0,0 +1,72 @@
+use tower::ServiceExt;
+
+use super::mempool::{unmined_transactions_in_blocks, Mempool};
+
+use tokio::sync::oneshot;
+use tower::builder::ServiceBuilder;
+
+use zebra_chain::{
+    parameters::Network,
+    transaction::{UnminedTx, UnminedTxId},
+};
+use zebra_consensus::Config;
+use zebra_network::{Request, Response};
+
+#[test]
+fn mempool_requests_for_transaction_ids() {
+    let runtime = tokio::runtime::Builder::new_current_thread()
+        .enable_all()
+        .build()
+        .expect("Failed to create Tokio runtime");
+    let _guard = runtime.enter();
+
+    let network = Network::Mainnet;
+    let config = Config::default();
+
+    let state_service = zebra_state::init_test(network);
+    let mut mempool_service = Mempool::new(network);
+
+    let added_transaction_ids: Vec<UnminedTxId> =
+        add_some_stuff_to_mempool(&mut mempool_service, network)
+            .iter()
+            .map(|t| t.id)
+            .collect();
+
+    runtime.block_on(async move {
+        let (chain_verifier, _transaction_verifier) =
+            zebra_consensus::chain::init(config.clone(), network, zebra_state::init_test(network))
+                .await;
+        let (_setup_tx, setup_rx) = oneshot::channel();
+
+        let inbound_service =
+            ServiceBuilder::new()
+                .load_shed()
+                .buffer(20)
+                .service(super::Inbound::new(
+                    setup_rx,
+                    state_service.clone(),
+                    chain_verifier.clone(),
+                    mempool_service,
+                ));
+
+        let request = inbound_service
+            .oneshot(Request::MempoolTransactionIds)
+            .await;
+        match request {
+            Ok(Response::TransactionIds(response)) => assert_eq!(response, added_transaction_ids),
+            _ => panic!("in this test we are pretty sure there is always a correct response"),
```

Comment:
TODO: change this message.

Reply:
https://github.com/ZcashFoundation/zebra/pull/2720/commits/eff862573655626dee2c6cb866f4574ff0d2e11c